File: CSharp part2/01. Arrays/11.BinarySearch/BinarySearch.cs
// Write a program that finds the index of given element in a sorted array of integers by using the Binary search algorithm.

using System;

class BinarySearch
{
    public static int Binary_Search(int[] arr, int x)
    {
        int mid = 0, low = 0, high = arr.Length - 1;
        while (low <= high)
        {
            mid = low + (high - low) / 2;
            if (x == arr[mid])
            {
                return mid;
            }
            else if (x < arr[mid])
                high = mid - 1;
            else
                low = mid + 1;
        }
        return -1;
    }
    static void Main()
    {
        int[] array = { 11, 9, 8, 6, 4};
        Array.Sort(array);
        foreach (int i in array)
        {
            Console.Write(i + ",");
        }
        Console.WriteLine();
        Console.Write("Enter an element from the array above:");
        int x = int.Parse(Console.ReadLine());
        bool found = Binary_Search(array, x) != -1;
        if (found)
        {
            Console.WriteLine("Element {0} is at position {1}.", x, Binary_Search(array, x));
        }
        else
        {
            Console.WriteLine("Element {0} was not found.", x);
        }
    }
}

File: JavaScript-Fundamentals/Arrays/07.BinarySearch/BinarySearch.js
// Write a script that finds the index of given element in a sorted array of integers by using the binary search algorithm.

var jsConsole,
    arr = [0, 5, 17, 26, 38, 45, 90],
    button = document.getElementById('check'),
    targetNumber,
    startIndex = 0,
    endIndex = arr.length,
    currIndex;

button.onclick = getIndex;

jsConsole.writeLine("Sample array is: [" + arr + "]");
console.log("Sample array is: [" + arr + "]");

function getIndex() {
    targetNumber = jsConsole.readInteger('#target_number');

    while ((endIndex - startIndex) >= 1) {
        currIndex = startIndex + (((endIndex - startIndex) / 2) | 0);

        if (arr[currIndex] === targetNumber) {
            jsConsole.writeLine(targetNumber + ' is at position ' + currIndex);
            console.log(targetNumber + ' is at position ' + currIndex);
            return;
        } else if (arr[currIndex] > targetNumber) {
            endIndex = currIndex;
        } else if (arr[currIndex] < targetNumber) {
            startIndex = currIndex;
        }
    }

    jsConsole.writeLine("Number was not found!");
    console.log("Number was not found!");
}
